akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -91,9 +81,57 @@ private[projection] class SlickProjectionImpl[Offset, StreamElement, P <: JdbcPr
       sourceProvider(offsetOpt)
     }
 
+    val handlerFlow: Flow[StreamElement, Done, _] =
+      strategy match {
+        case ExactlyOnce =>
+          Flow[StreamElement]
+            .mapAsync(1)(processElementAndStoreOffset)
+
+        case AtLeastOnce(1, _) =>
+          Flow[StreamElement]
+            .mapAsync(1) { elt => processElement(elt).flatMap(_ => storeOffset(offsetExtractor(elt))) }
+
+        case AtLeastOnce(afterElements, orAfterDuration) =>
+          Flow[StreamElement]
+            .map(elt => offsetExtractor(elt) -> elt)
+            .mapAsync(1) {
+              case (offset, element) => processElement(element).map(_ => offset)
+            }
```

Comment:
Why creating a tuple? 

```suggestion
            .mapAsync(1) { elt => 
              processElement(elt).map(_ => offsetExtractor(elt))
            }
```

Reply:
For slick, we can take the two DBIOs, combine them and run without using `transactionally`. 

It may give a small performance improvements as we take one thread from Slick thread pool, one connection and execute two SQL statements each in its own transaction.

Reply:
We will need to test the following though:

User returns in event handler:

`dbio1.flatMap(_ => dbio2).transactionaly`

we do:

```scala
userHandlerDbio // <- the one above
  .flatMap(_ => offsetStore.saveOffset(projectionId, offset))
```

I'm 99% sure, but we need to confirm this behavior. If I recall correctly, user DBIO will be run in one transaction and offset store in another. That would be obvious, but I want to confirm this.